Fix: Added Rebuild button for function view when build fails during deployment

singhbhaskar opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.

You can learn more about contributing to appwrite here: https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md

Happy contributing!

-->

## What does this PR do?

Adds a new button to rebuild function deployment if it fails. The button is not showed when the build is successfull.

## Test Plan

Mannual testing. Ran `npm run build`, `npm run format` and `npm run lint`

## Related PRs and Issues

https://github.com/appwrite/appwrite/issues/4840

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
Yes I have read the contributing guidelines.

stnguyen90 commented on src/routes/console/project-[project]/functions/function-[function]/rebuild.svelte:
```diff
@@ -0,0 +1,46 @@
+    <p>Are you sure you want to re-build this deployment?</p>
+    <svelte:fragment slot="footer">
+        <Button text on:click={() => (showRebuild = false)}>Cancel</Button>
+        <Button secondary submit>Re-build</Button>
```
```suggestion
        <Button secondary submit>Retry build</Button>
```

singhbhaskar replied:
Addressed it in the latest commit.

stnguyen90 commented on src/routes/console/project-[project]/functions/function-[function]/rebuild.svelte:
```diff
@@ -0,0 +1,46 @@
+            showRebuild = false;
+            addNotification({
+                type: 'success',
+                message: `Deployment has been re-build`
```
This can be:

```suggestion
                message: `Retrying build`
```

singhbhaskar replied:
Addressed it in the latest commit.

stnguyen90 reviewed (changes requested):
This looks great! Good idea on the confirmation modal too! I just have some minor text suggestions.

singhbhaskar commented:
> This looks great! Good idea on the confirmation modal too! I just have some minor text suggestions.

Sure Steven, I'll update the suggest changes. Thanks

TorstenDittmann commented on src/routes/console/project-[project]/functions/function-[function]/+page.svelte:
```diff
@@ -223,6 +229,17 @@
                                                 }}>
                                                 Activate
                                             </DropListItem>
+                                            {#if ['failed'].includes(deployment.status)}
```
```suggestion
                                            {#if 'failed' === deployment.status}
```

singhbhaskar replied:
Addressed it in the latest commit.

singhbhaskar commented on src/routes/console/project-[project]/functions/function-[function]/rebuild.svelte:
```diff
@@ -0,0 +1,46 @@
+
+<Modal bind:show={showRebuild} onSubmit={handleSubmit}>
+    <svelte:fragment slot="header">Re-build Deployment</svelte:fragment>
+    <p>Are you sure you want to re-build this deployment?</p>
```
Addressed it in the latest commit.

singhbhaskar commented:
Hi @stnguyen90 ,
I have addressed all the comments. Kindly review again. Attaching the latest screenshots.

### Screenshot for function failure:
<img width="1431" alt="retryBuildOptions" src="https://user-images.githubusercontent.com/20492520/232302951-3701c3d4-2921-4b59-882b-33dcf1c00e7c.png">

### Screenshot for Model Alert when user clicks on rebuild button:
<img width="1431" alt="retryBuildModel" src="https://user-images.githubusercontent.com/20492520/232303249-a7250208-8de8-47f6-a54c-304827d86636.png">

### Screenshot for Notification alert when rebuild request is process by the SDK:
<img width="1431" alt="retryBuildSucess" src="https://user-images.githubusercontent.com/20492520/232303344-ebd460fb-5a9a-474d-b010-fcda82647abe.png">

stnguyen90 commented:
@singhbhaskar looks like the tests are failing. Perhaps the package name has changed?

Also, please make sure to re-request a review when you're ready for a review.

singhbhaskar commented:
Hi @stnguyen90 ,
Updated the code to fix test. Please review.

stnguyen90 commented on src/routes/console/project-[project]/functions/function-[function]/rebuild.svelte:
```diff
@@ -0,0 +1,46 @@
+</script>
+
+<Modal bind:show={showRebuild} onSubmit={handleSubmit}>
+    <svelte:fragment slot="header">Retry build Deployment</svelte:fragment>
+    <p>Are you sure you want to retry build this deployment?</p>
```
Not quite right. Please update to:

```suggestion
    <svelte:fragment slot="header">Retry build</svelte:fragment>
    <p>Are you sure you want to retry building this deployment?</p>
```

singhbhaskar commented:
I am able to successfully run lint and test in my local system but for some reason it's failing during the check. Can you trigger a rebuild?

stnguyen90 commented:
> I am able to successfully run lint and test in my local system but for some reason it's failing during the check. Can you trigger a rebuild?

@TorstenDittmann, I also tried to check the linting and it seems fine.

I manually ran the tests and they all passed:

<img width="583" alt="image" src="https://user-images.githubusercontent.com/1477010/235809086-040d853a-3c6e-4df2-95d4-ffd8a141b96e.png">

I can confirm the modal triggers a new build:

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/1477010/235810857-cd4d071d-9ed9-4438-9e74-a3d396dc31fa.png">

What do you think about just merging?

TorstenDittmann reviewed (changes requested):
once you have fixed the formatting issues we can merge 👍🏻 

just run `npm run format`

singhbhaskar commented:
> npm run format

Hi,
I had already ran `npm run format` when raising the PR. I re-ran it just now and there is no file format change for any file. Can you tell which file is having format issue?
